src/app/pages/map-view/sidebar-section/layer-card/param-item/param-item.component.html

```diff
@@ -0,0 +1,31 @@
+<!-- <ng-container *ngIf="type==='number'"> -->
+<label class="checkbox-label" [class.-inactive]="check.checked" [class.-hide]="type!=='number'">
+        <input #check class="checkbox" type="checkbox" (click)="neglectParam(desc.value, num.value)">                       
+        <span class="icon"></span>
+      </label>
+<select #desc class="desc" [class.-hide]="type!=='number'" (change)="updateValue(desc.value, num.value)">
+        <option value="1" [selected]="value>0">Near</option>
+        <option value="-1" [selected]="value<0">Far</option>
+      </select>
+<!-- </ng-container> -->
+
+<span class="name">{{param | rejoin:"_" | titlecase}}</span>
+<select #num [class.-hide]="type!=='number'" class="num" (change)="updateValue(desc.value, num.value)">
+  <option *ngFor="let val of genRange(range)" [value]="val" [selected]="val===this.abs(this.value)">{{val}}</option>
+</select>
+
+<ng-container [class.-hide]="type!=='string'">
+    <select *ngIf="value=='tin' || value=='idw'" #string class="string" (change)="updateValue(string.value)">
```

Comment:
Watch out for these magic string constants. You've broken the abstraction that you only need to define a new `LayerCard` and pass it to the appropriate components for it to fully work correctly, since you've now coupled this display to strings defined in your layer card configuration block. 

You might be able to add a values => label mapping to your layer card config and pull the values/labels from there to avoid this coupling. All of the if blocks here look very similar.